packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -162,7 +185,28 @@ export const plugin = <TContext>(
       let didResolveSource: boolean = false;
 
       return {
+        async didResolveOperation(requestContext) {
+          if (typeof options.traceReporting !== 'function') return;
+          const shouldReportTrace = await options.traceReporting(
+            requestContext,
+          );
+
+          // Help the user understand they've returned an unexpected value,
+          // which might be a subtle mistake.
+          if (typeof shouldReportTrace !== 'boolean') {
+            (requestContext.logger || logger).warn(
+              "The 'traceReporting' predicate function must return a boolean value.",
+            );
+            return;
+          }
+
+          if (!shouldReportTrace) didEnd(requestContext, false);
+        },
         didResolveSource(requestContext) {
+          // We can early abort if traceReporting returned false
```

Comment:
This case will never happen, because didResolveSource happens before didResolveOperation.

(Oh, you should probably put them in that order in the code!)

Reply:
Fixed

Reply:
I don't see any changes associated with this or other comments